pkg/tsdb/postgres/macros.go

```diff
@@ -30,8 +28,6 @@ func newPostgresMacroEngine(timescaledb bool) sqleng.SQLMacroEngine {
 
 func (m *postgresMacroEngine) Interpolate(query plugins.DataSubQuery, timeRange plugins.DataTimeRange,
 	sql string) (string, error) {
-	m.timeRange = timeRange
```

Comment:
So all of this concurrent map writes panic was about we share the timeRange from request to all subrequest, and each subrequest is trying to assign/read time range and query from postgresMacroEngine? :)

Reply:
No, the problem I believe was we're making changes to `m.query.Model` and m.query could be overwritten in case of concurrent request making multiple concurrent queries trying to write to the same map and hence the panic.